tests/fixtures/git_repo.py

```diff
@@ -31,17 +48,18 @@ def example_git_https_url():
     # twice, once with a different URL format
     params=[lazy_fixture("example_git_ssh_url")]
 )
-def git_repo_factory(request, example_project):
-    """
-    !!! WARNING !!!
-    You must call repo.close() after yield-ing the result of
-    calling this factory in a test, otherwise the test suite will fail with
-    OSError: Too Many Open Files
-    See https://github.com/pytest-dev/pytest/issues/2970#issuecomment-348033023
-    """
-
-    def git_repo():
+def git_repo_factory(
+    request: pytest.FixtureRequest,
+    example_project: ExProjectDir
+) -> Generator[RepoInitFn, None, None]:
+    repos: list[Repo] = []
+
+    def git_repo() -> Repo:
         repo = Repo.init(example_project.resolve())
+
+        # store the repo so we can close it later
+        repos.append(repo)
```

Comment:
👍 this is neat, though it will keep a reference to `Repo` instances for longer than they used to exist. I don't see that being a problem, but if it ever is then a `weakref.WeakSet` should help